src/wallet/rpcwallet.cpp

```diff
@@ -93,7 +95,18 @@ void WalletTxToJSON(const CWalletTx& wtx, UniValue& entry)
         entry.push_back(Pair("blockindex", wtx.nIndex));
         entry.push_back(Pair("blocktime", mapBlockIndex[wtx.hashBlock]->GetBlockTime()));
         entry.push_back(Pair("expiryheight", (int64_t)wtx.nExpiryHeight));
+        status = "mined";
```

Comment:
Should this status be mutually exclusive with the others?  Can a `coinbase` transaction be "waiting"?

Reply:
Thank you for the comment and please excuse my ignorance. Do you think we should add a new type `generated` for when the transaction is coinbase ignoring any of the other statuses ?

Reply:
I don't yet have an adequately informed opinion to answer intelligently.

Reply:
It's technically possible to have a coinbase transaction in the wallet that is not mined, if a mining pool mines a block that is then orphaned. I believe it would show up here as `waiting`, so there may be value in having a separate `orphaned` state, but that can be added later if necessary. In any case, `generated` is not a status we want, as there is already a `generated` property marking coinbase, and coinbase transactions can have multiple statuses.